diff -urNp a/dnf/base.py b/dnf/base.py
--- a/dnf/base.py	2019-11-05 17:26:35.000000000 +0800
+++ b/dnf/base.py	2020-01-17 09:10:06.178837683 +0800
@@ -442,7 +442,7 @@ class Base(object):
             logger.info(_("The downloaded packages were saved in cache "
                           "until the next successful transaction."))
             logger.info(_("You can remove cached packages by executing "
-                          "'%s'."), "{prog} clean packages").format(prog=dnf.util.MAIN_PROG)
+                          "'%s'."), "{prog} clean packages".format(prog=dnf.util.MAIN_PROG))
 
         # Do not trigger the lazy creation:
         if self._history is not None:
