From 22106c470fe0decced1df17cb2b7b25c13ad3519 Mon Sep 17 00:00:00 2001
From: Daniel Mach <dmach@redhat.com>
Date: Wed, 12 Dec 2018 14:36:12 +0100
Subject: [PATCH 105/175] [callback] Bring PKG_ERASE back for compatibility
 reasons.

Resolves: rhbz#1597182
Signed-off-by: root <root@localhost.localdomain>
---
 dnf/callback.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dnf/callback.py b/dnf/callback.py
index 967fb7a7..4b162fcf 100644
--- a/dnf/callback.py
+++ b/dnf/callback.py
@@ -28,6 +28,7 @@ PKG_INSTALL = dnf.transaction.PKG_INSTALL  # :api
 PKG_OBSOLETE = dnf.transaction.PKG_OBSOLETE  # :api
 PKG_REINSTALL = dnf.transaction.PKG_REINSTALL  # :api
 PKG_REMOVE = dnf.transaction.PKG_ERASE  # :api
+PKG_ERASE = PKG_REMOVE  # deprecated, use PKG_REMOVE instead
 PKG_UPGRADE = dnf.transaction.PKG_UPGRADE  # :api
 
 PKG_CLEANUP = dnf.transaction.PKG_CLEANUP  # :api
-- 
2.19.1

